# We are conducting a research on registration fees for scientific conferences in Nigeria. 

In [41]:
import math
import scipy.stats

### A. Estimating Sample of Healthcare Providers

- Estimating the sample size needed for finding the mean attendance rate of healthcare providers that is statistically significant
- We are working with:
    - a population size of 353,600 health professionals in Nigeria
    - a 95% confidence level 
    - a 5% margin of error
    - an estimated variance of 60%

In [42]:
def calculate_sample_size(z_score, proportion, margin_of_error):
    # Calculate the sample size
    sample_size = ((z_score**2) * proportion * (1 - proportion)) / (margin_of_error**2)
    
    return math.ceil(sample_size)  # Round up to the nearest whole number

confidence_level = 0.95  # 95% confidence level
z_score = 1.96  # Z-score for 95% confidence
estimated_proportion = 0.6  # 60% healthcare professionals attend scientific conferences
margin_of_error = 0.05  # 5% margin of error

sample_size = calculate_sample_size(z_score, estimated_proportion, margin_of_error)

In [43]:
print(sample_size)

369


### B. Estimating Sample Size of Conferences
- We want to estimate the sample size needed for finding the mean registration fee that is statistically significant
- We are working with:
    - a 95% confidence level 
    - a 10% margin of error
    - an estimated 25% of yearly conferences in Nigeria being healthcare conferences

In [44]:
def calculate_sample_size(z_score, proportion, margin_of_error, population_size):
    # Calculate the sample size without finite population correction
    n_0 = ((z_score**2) * proportion * (1 - proportion)) / (margin_of_error**2)
    
    # Apply finite population correction
    conference_sample_size = (population_size * n_0) / (population_size + n_0 - 1)
    
    return math.ceil(conference_sample_size)  # Round up to the nearest whole number

confidence_level = 0.95  # 95% confidence level
z_score = 1.96  # Z-score for 95% confidence
estimated_proportion = 0.25  # 25% of scientific conferences are healthcare related (estimated)
margin_of_error = 0.1  # 10% margin of error
total_population = 1200  # Total population size

conference_sample_size = calculate_sample_size(z_score, estimated_proportion, margin_of_error, total_population)

In [45]:
print(conference_sample_size)

69


### C. Estimating Sample Size of registration fees

- We want to estimate the sample size needed for finding the mean registration fee that is statistically significant
- We are working with:
    - a 95% confidence level 
    - a ₦5,000 margin of error
    - an estimated ₦20,000 standard deviation

In [46]:
margin_of_error = 5000   #Desired margin of error
std_dev = 20000   #Population standard deviation (an estimate was used)

#Calculate the Z-score for the confidence level
z_score = scipy.stats.norm.ppf(1 - (1 - confidence_level) / 2)

#Calculate the sample size
registration_sample_size = ((z_score * std_dev) / margin_of_error) ** 2

#Round up to the nearest whole number
registration_sample_size = math.ceil(registration_sample_size)

In [47]:
print(registration_sample_size)

62


Thus, 
The questionnaires need to reach a minimum of 369 healthcare professionals and cover at least 69 different conferences across the country.